tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpConnectionIT.java

```diff
@@ -89,13 +89,19 @@ public void testConnectSucceedsForRegisteredDevice(final TestContext ctx) {
     @Test
     public void testConnectFailsForNonExistingDevice(final TestContext ctx) {
 
-        // GIVEN an adapter
-        // WHEN an unknown device tries to connect
-        connectToAdapter(IntegrationTestSupport.getUsername("non-existing", Constants.DEFAULT_TENANT), "secret")
-        .setHandler(ctx.asyncAssertFailure(t -> {
-            // THEN the connection is refused
-            ctx.assertTrue(t instanceof SaslException);
-        }));
+        // GIVEN an existing tenant
+        final String tenantId = helper.getRandomTenantId();
+        final TenantObject tenant = TenantObject.from(tenantId, true);
+
+        helper.registry.addTenant(JsonObject.mapFrom(tenant)).compose( r ->
```

Comment:
what purpose does it serve to register the random tenant when it is not used later on?

Reply:
if no corresponding tenant exists, the adapters is blocked to connect, and cannot check if the device exists.

Reply:
that's true, but then you should also use that tenant in the device credentials later on (instead of DEFAULT_TENANT), right?